libs/libc/uuid/lib_uuid_create.c

```diff
@@ -42,17 +43,24 @@
 
 void uuid_create(uuid_t *u, uint32_t *status)
 {
-#ifdef CONFIG_CRYPTO_RANDOM_POOL
-  arc4random_buf(u, sizeof(uuid_t));
-#else
-  unsigned long *beg = (unsigned long *)u;
-  unsigned long *end = (unsigned long *)(u + 1);
+  ssize_t ret;
 
-  while (beg < end)
+  ret = getrandom(u, sizeof(uuid_t), 0);
+  if (ret < 0)
     {
-      *beg++ = rand();
+      ret = getrandom(u, sizeof(uuid_t), GRND_RANDOM);
```

Comment:
Do we need to check the EINT and repeat again in case if call was interrupted by a signal?

Reply:
Done.